File: server/php/test/hz/message/message_test.php
<?php

use PHPUnit\Framework\TestCase;
require_once('../../../src/hz/message/message.php');

/**
 * 测试message.Message
 */
class test_Message extends TestCase {
	/** message对象 */
	private $message;
	/** post对象 */
	private $post;
	
	/**
	 * @before
	 */
	public function before() {
        $holder = new TestHolder();
        $this->post = new TestPost();
        $this->message = new Message($holder, $this->post);
	}
	
	/**
	 * @after
	 */
	public function after() {
        $this->message->close();
	}
	
	/**
	 * 测试message.Message.request()
	 */
	public function test_request() {
        // 测试['type'=>'get_post_count']
        $res = $this->message->request(['type'=>'get_post_count']);
        $this->assertEquals($res, $this->post->getCount());

        // 测试['type'=>'get_all_post']
        $res = $this->message->request(['type'=>'get_all_post']);
        $this->assertEquals($res, $this->post->getAll());

        // 测试['type'=>'get_post_by_id']
        $res = $this->message->request(['type'=>'get_post_by_id']);
        $this->assertEquals($res, $this->post->getById([]));

        // 测试['type'=>'add_post']
        $res = $this->message->request(['type'=>'add_post']);
        $this->assertEquals($res, $this->post->add([]));

        // 测试['type'=>'modify_post']
        $this->message->request(['type'=>'modify_post']);
        $this->assertEquals($res, $this->post->modify([]));

        // 测试['type'=>'remove_post']
        $res = $this->message->request(['type'=>'remove_post']);
        $this->assertEquals($res, $this->post->remove([]));

        // 测试['type'=>'other']
        $res = $this->message->request(['type'=>'other']);
        $this->assertEquals($res['status'], 'invalid_parameter');
	}
}

/**
 * 模拟holder类
 */
class TestHolder {
	public function destroy() {
	}
	
	public function close() {
	}
}

/**
 * 模拟post类
 */
class TestPost {
	public function getCount() {
        return ['status'=>'ok', 'data'=>['count'=>1234]];
	}

	public function getAll() {
        return ['status'=>'ok', 'data'=>[['id'=>1234, 'timestamp'=>123456, 'content'=>'abcd'], ['id'=>12345, 'timestamp'=>1234567, 'content'=>'abcde']]];
	}

	public function getById($data) {
        return ['status'=>'ok', 'data'=>['id'=>1234, 'timestamp'=>123456, 'content'=>'abcd']];
	}

	public function add($data) {
        return ['status'=>'ok', 'data'=>['id'=>1234, 'timestamp'=>123456, 'content'=>'abcd']];
	}

	public function modify($data) {
        return ['status'=>'ok', 'data'=>['id'=>1234, 'timestamp'=>123456, 'content'=>'abcd']];
	}

	public function remove($data) {
        return ['status'=>'ok', 'data'=>['id'=>1234]];
	}
}

?>

File: server/python/test/hz/message/message_test.py
#!/usr/bin/env python
#-*-coding: utf-8-*-

'''测试message'''

import os, sys
sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), '../../../src/'))
from hz.message.message import Message
import unittest

class test_Message(unittest.TestCase):
    '''测试message.Message'''
    def setUp(self):
        holder = TestHolder()
        self.__post = TestPost()
        self.__message = Message(holder, self.__post)
        
    def tearDown(self):
        self.__message.close();
    
    def test_request(self):
        '''测试message.Message.request()'''
        # 测试{'type': 'get_post_count'}
        self.assertEqual(self.__message.request({'type': 'get_post_count'}), self.__post.getCount())
        
        # 测试{'type': 'get_all_post'}
        self.assertEqual(self.__message.request({'type': 'get_all_post'}), self.__post.getAll())
        
        # 测试{'type': 'get_post_by_id'}
        self.assertEqual(self.__message.request({'type': 'get_post_by_id'}), self.__post.getById({}))
        
        # 测试{'type': 'add_post'}
        self.assertEqual(self.__message.request({'type': 'add_post'}), self.__post.add({}))
        
        # 测试{'type': 'modify_post'}
        self.assertEqual(self.__message.request({'type': 'modify_post'}), self.__post.modify({}))
        
        # 测试{'type': 'remove_post'}
        self.assertEqual(self.__message.request({'type': 'remove_post'}), self.__post.remove({}))
        
        # 测试{'type': 'other'}
        self.assertEqual(self.__message.request({'type': 'other'})['status'], 'invalid_parameter')

class TestHolder:
    '''模拟holder类'''
    def inst(self):
        return None

    def close(self):
        return None

    def destroy(self):
        return None

class TestPost:
    '''模拟post类'''
    def getCount(self):
        return {'status': 'ok', 'data': {'count': 1234}}

    def getAll(self):
        return {'status': 'ok', 'data': [{'id': 1234, 'timestamp': 123456, 'content': 'abcd'}, {'id': 12345, 'timestamp': 1234567, 'content': 'abcde'}]}

    def getById(self, data):
        return {'status': 'ok', 'data': {'id': 1234, 'timestamp': 123456, 'content': 'abcd'}}

    def add(self, data):
        return {'status': 'ok', 'data': {'id': 1234, 'timestamp': 123456, 'content': 'abcd'}}

    def modify(self, data):
        return {'status': 'ok', 'data': {'id': 1234, 'timestamp': 123456, 'content': 'abcd'}}

    def remove(self, data):
        return {'status': 'ok', 'data': {'id': 1234}}

if __name__ == '__main__':
    unittest.main()
